plugins/c8y_firmware_plugin/src/firmware_manager.rs

```diff
@@ -598,16 +607,22 @@ pub struct FirmwareOperationEntry {
 }
 
 impl FirmwareOperationEntry {
-    fn create_file(&self) -> Result<(), FirmwareManagementError> {
-        let path = PersistentStore::get_file_path(&self.operation_id);
+    fn create_file(
```

Comment:
It feels more like this `create_file` or rather `create_operation_file` and `overwrite_file` functions should have been part of the `PresistentStore` as creating a file in the store should have been a function of that store.